store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/FederatedAddGraphHandlerParent.java

```diff
@@ -93,6 +98,20 @@ public Void doOperation(final OP operation, final Context context, final Store s
         return null;
     }
 
+    private static void processNonViableFedStoreWithNonPersistentCaches(final GraphSerialisable graphSerialisable) throws OperationException {
+        final String storeClass = graphSerialisable.getStoreProperties().getStoreClass();
+        final Class<? extends Store> aClass;
+        try {
+            aClass = Class.forName(storeClass).asSubclass(Store.class);
+        } catch (ClassNotFoundException e) {
+            throw new GafferRuntimeException("Error creating Store class from properties", e);
+        }
+        if (FederatedStore.class.isAssignableFrom(aClass) && CacheServiceLoader.getService() instanceof HashMapCacheService) {
```

Comment:
Maybe the various cache services could have a trait like `isPersistent` or something?

Reply:
Yes that was my thinking.